addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/CsrfCountermeasuresScanRule.java

```diff
@@ -160,6 +155,12 @@ public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
                         }
                     }
                 }
+                String formMethod = formElement.getAttributeValue("method");
+                formMethod = formMethod == null ? HttpRequestHeader.GET : formMethod;
+                if (!AlertThreshold.LOW.equals(getAlertThreshold())
+                        && HttpRequestHeader.GET.equals(formMethod)) {
+                    continue;
+                }
```

Comment:
This simpler/faster check should be before others, there's no point doing extra checks if we are going to skip it.

Reply:
The `numberOfFormsPassed` should still be increased, I guess we don't have enough tests covering that.

Reply:
Addressed

Reply:
Grr maybe not, gimme a few

Reply:
Okay, the test now asserts that the count is correct where the value is used.